routing_common/transit_types.hpp

```diff
@@ -104,20 +104,16 @@ class Gate
   m2::PointD const & GetPoint() const { return m_point; }
 
   DECLARE_VISITOR_AND_DEBUG_PRINT(Gate, visitor(m_featureId, "osm_id"),
-                                  visitor(m_entrance, "entrance"),
-                                  visitor(m_exit, "exit"), visitor(m_weight, "weight"),
-                                  visitor(m_stopIds, "stop_ids"), visitor(m_point, "point"))
+                                  visitor(m_pedestrianFeatureIds, "" /* name */),
```

Comment:
Does this field needs visiting at all? And where is the visitor for "gates"?

Reply:
This fields need to be visited with routing::transit::Serializer/Deserializer at routing_commont/transit_serdes.hpp and with DebugPrint. It doesn't need to be visited with DeserializerFromJson()

Reply:
Why did you skip name for m_predestrianFeatureIds?

Reply:
Because the name is not used. There's no field for |m_predestrianFeatureIds| at json. |m_predestrianFeatureIds| is generated while transit section generation based on gate (exit or entrance) coords.

Reply:
I decided not to skip field name because of debug print.

Reply:
It looks like that it's a visitor should decide what fields to skip and what fields not to skip. Missing the field name is generally a bad idea, becase DECLARE_VISITOR declares a generic Visit function that can be used with any visitor, not only with json.

Reply:
Therefore I dislike the idea to skip the name for m_pedestrianFeatureIds. If you don't have a field in json, please, modify your json visitor to skip this field.

Reply:
Now the field is not visited with my json visitor because there's no such field in the json. So the field has a default value (it's a empty vector) after deserialization from json. 

> If you don't have a field in json, please, modify your json visitor to skip this field.

How do you propose to modify my json visitor to skip the field? Based on field name?